### PR TITLE
Blr/statham/content type

### PR DESCRIPTION
**Pull request by bartriordan**
@BJK @maneeshnycda Please review.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
 import ActionPrototype from './lib/ActionPrototype'
+import AjaxAdaptorBase from './lib/AjaxAdaptorBase'
```
Comment (bartriordan) [inline src/index.js:2]: Exporting this so we can extend it in `web-apps`

File: src/lib/StorePrototype.js [hunk 2]
```diff
@@ -15,80 +15,64 @@ import AppDispatcher from './fu-dispatcher'
 
 function factory(registerAction, extendType, registerMessageAction) {
   var StorePrototype = assign(EventEmitter.prototype, {
```
Comment (bartriordan) [inline src/lib/StorePrototype.js:17]: This file was a mess. I cleaned it up and organized the methods so it looks better to outsiders

File: src/lib/jsonStatham.js [hunk 3]
```diff
@@ -20,18 +20,19 @@ function ensureDataIsObject(data) {
 var jsonStatham = {
   setAdaptor(adaptor) { this.adaptor = adaptor },
   getAdaptor() { return (this.adaptor ? this.adaptor : new defaultAdaptor()) },
-  buildRequest(path, method, data, withCredentials, additionalHeaders) {
+  buildRequest(path, method, data, additionalHeaders) {
```
Comment (bartriordan) [inline src/lib/jsonStatham.js]: Killed the `withCredentials`

File: src/lib/jsonStatham.js [hunk 4]
```diff
@@ -51,14 +52,14 @@ var jsonStatham = {
...
   put(url, data) { return this.sendRequest(this.buildRequest(url, 'PUT', data)) },
-  post(url, data, withCredentials) { return this.sendRequest(this.buildRequest(url, 'POST', data, withCredentials)) },
-  delete(url, data, withCredentials) { return this.sendRequest(this.buildRequest(url, 'DELETE', data, withCredentials)) },
+  post(url, data) { return this.sendRequest(this.buildRequest(url, 'POST', data)) },
+  delete(url, data) { return this.sendRequest(this.buildRequest(url, 'DELETE', data)) },
   postFile(url, data) {
     // Taken from: http://stackoverflow.com/questions/12431760/html5-formdata-file-upload-with-rubyonrails
     // and http://stackoverflow.com/questions/21234106/upload-file-using-reactjs-via-blueimp-fileupload-jquery-plugin
-    var request = this.buildRequest(url, 'POST', data)
+    var request = this.buildRequest(url, 'POST', data, {contentType: false})
```
Comment (bartriordan) [inline src/lib/jsonStatham.js]: Override the default `contentType` for just this request type

**Comment by bartriordan** [comment]
@BJK @maneeshnycda This should wait until I get a PR up to remove all usage of `withCredentials` in `web-apps`

**Comment by maneeshanand** [comment]
👍  looks good. let me know when `web-apps` is ready for this.

**Comment by bartriordan** [comment]
This should be merged with https://github.com/everplans/web-apps/pull/702

**Comment by maneeshanand** [comment]
👍  looks good to me

**Pull request merged**